src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts

```diff
@@ -97,54 +104,50 @@ export class CheckboxSelectColumn<T extends Slick.SlickData> implements Slick.Pl
 		this._grid.render();
 
 		if (!this._options.title) {
-			if (selectedRows.length && selectedRows.length === this._grid.getDataLength()) {
-				this._grid.updateColumnHeader(this._options.columnId!,
-					strings.format(checkboxTemplate, 'checked', this.getAriaLabel(true)),
-					this._options.toolTip);
-			} else {
-				this._grid.updateColumnHeader(this._options.columnId!,
-					strings.format(checkboxTemplate, '', this.getAriaLabel(false)),
-					this._options.toolTip);
-			}
+			// when no title is specified, show the select all/deselect all checkbox
```

Comment:
is there an option for the select all/deselect all checkbox be there even if the column has a title?

Reply:
it is possible to add it, but that is not in the scope of this PR, I am just refactoring the code here, the behavior remains the same.